### PR TITLE
Added documentation for Tessera migration tool

### PR DESCRIPTION
**Pull request by vmichalik**
## PR description
Adds documentation for the Orion-to-Tessera migration tool.

Related to hyperledger/besu#1953

### CODE REVIEW

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (macfarla) [inline docs/Tutorials/Migrating-to-Tessera.md]: for help with other databases.

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (macfarla) [inline docs/Tutorials/Migrating-to-Tessera.md]: s/will be output/will match expected values

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (macfarla) [inline docs/Tutorials/Migrating-to-Tessera.md]: if you want to use these to verify state, ideally you've already run these calls against the orion before you shut it down. So maybe add an extra step for that

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (macfarla) [inline docs/Tutorials/Migrating-to-Tessera.md]: did this work for you? I was still building it from the tessera dir

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (macfarla) [inline docs/Tutorials/Migrating-to-Tessera.md]: the config for Besu doesn't change at all, but not sure how to make that clear in a concise way

Comment (NicolasMassart) [inline docs/Tutorials/Migrating-to-Tessera.md]: I'm not sure there's really a requirement to start Tessera before Besu but if there's one, maybe:
```suggestion
1. Start Tessera with the new config and database files
1. Start Hyperledger Besu nodes
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [reply docs/Tutorials/Migrating-to-Tessera.md]: Yes building from the subdir works for me too.

Comment (vmichalik) [reply docs/Tutorials/Migrating-to-Tessera.md]: It did work for me at one point - and this follows the instructions on the commit, but after all my problems building on the M1 arch I've asked @NicolasMassart to help test!

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (vmichalik) [reply docs/Tutorials/Migrating-to-Tessera.md]: attempted a resolution

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (vmichalik) [reply docs/Tutorials/Migrating-to-Tessera.md]: added

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (vmichalik) [reply docs/Tutorials/Migrating-to-Tessera.md]: changed

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (vmichalik) [reply docs/Tutorials/Migrating-to-Tessera.md]: added

**Review by vmichalik — COMMENTED** [review]
doc password protected keys, migrated and renamed

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-to-Tessera.md:53]: This is not the output of the latest version. Message was fixed and now is:
```
% ./bin/migrate --help
Usage: orion-to-tessera/bin/migrate [-h] -f=Orion config file -o=<outputFile>
                                    tessera.jdbc.password=<password> tessera.
                                    jdbc.url=<url> tessera.jdbc.user=<username>
  -f, orionfile, orionconfig=Orion config file
                     Orion config file
  -h, help, --help   Print this message
  -o, outputfile=<outputFile>
                     Output Tessera config file
      tessera.jdbc.password=<password>
                     Target Tessera DB password
      tessera.jdbc.url=<url>
                     Target Tessera DB JDBC connection string
      tessera.jdbc.user=<username>
                     Target Tessera DB username
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-to-Tessera.md:81]: Not sure what this script is. If you mean the script we just built it's `./bin/migrate` directly without `.bash`.
Otherwise I don't have the `bin` dir in my home (`~/`)

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-to-Tessera.md]: This is not the output of the latest version. Message was fixed and now is:
```
% ./bin/migrate --help
Usage: orion-to-tessera/bin/migrate [-h] -f=Orion config file -o=<outputFile>
                                    tessera.jdbc.password=<password> tessera.
                                    jdbc.url=<url> tessera.jdbc.user=<username>
  -f, orionfile, orionconfig=Orion config file
                     Orion config file
  -h, help, --help   Print this message
  -o, outputfile=<outputFile>
                     Output Tessera config file
      tessera.jdbc.password=<password>
                     Target Tessera DB password
      tessera.jdbc.url=<url>
                     Target Tessera DB JDBC connection string
      tessera.jdbc.user=<username>
                     Target Tessera DB username
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-to-Tessera.md]: Not sure what this script is. If you mean the script we just built it's `./bin/migrate` directly without `.bash`.
Otherwise I don't have the `bin` dir in my home (`~/`)

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-to-Tessera.md]: could you add links to the doc for these methods?
What result should we expect form them?

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-to-Tessera.md]: should `main class` be indicated here or simply `migrate`? See also in the next lines where the command template is displayed.

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-to-Tessera.md]: To match the command we used previously for `--help`
```suggestion
    ./bin/migrate -f <Orion config file> -o <outputFile> tessera.jdbc.password <password> tessera.jdbc.url <url> tessera.jdbc.user <username>
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-to-Tessera.md]: Use simpler paths. Just consider users are already in the right `build/install/migrate/` directory as you asked to run `./bin/migrate --help` previously.
Remove ambiguity on the option values (using the option name as a value for an example is not clear).
Don't hesitate to use `"` around values. It's not always necessary if there's no spaces in the value, but it's safer and users who don't know will have something that always works.

```suggestion
    ./bin/migrate -f "orion.conf" -o="tessera-migrated.conf" tessera.jdbc.password "My Secret Pass" tessera.jdbc.url "jdbc:h2:tessera1" tessera.jdbc.user "user1"
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [reply docs/Tutorials/Migrating-to-Tessera.md]: If you use the `gradle` installed on your computer, yes it works. If you don't have gradle, it requires that you use the wrapper with `./gradlew` from the root. This is probably what we should require for users.
The best of course would be that they don't have to build, but simply use binaries on a JRE or in a Docker image => explaining the build from source is maybe not targeted to the audience of this migration tool.

Anyway, the content of this doc seems to imply that you ran `gradle installDist` and not `gradle build` as you then use `./bin/migrate` directly which is only available in the `tessera/migration/orion-to-tessera/build/install/migrate` dir. (otherwise would require to untar the dist from `tessera/migration/orion-to-tessera/build/distribution/`.)

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-to-Tessera.md]: Same comment here about links to the methods.
Also instead of writing "results from earlier", clearly link to the previous section. You may need to split the `## Run migration` (or `##Migrate`) with sub-headers for that. Maybe `### Stop services`, `### Run migration` and `## Restart services` or alike.

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-to-Tessera.md]: Simplify and use sentence case.
```suggestion
## Migrate
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-to-Tessera.md]: I think it's not required and it will always be `migration`...and if not always the same, it should...

File: mkdocs.yml
Comment (NicolasMassart) [inline mkdocs.yml]:
```suggestion
        - Migrating from Orion to Tessera: Tutorials/Migrating-to-Tessera.md
```
and maybe rename the file to `Migrating-from-Orion-to-Tessera.md` too.

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-to-Tessera.md]: `./gradlew` alone is is not building if you don't tell it what task to run.
Also if you run `./gradlew clean build` as the equivalent of the previously documented command but with the wrapper, then all tessera will build and test and we only need to build the `orion-to-tessera` module.
Plus it's not going to install the dist, meaning you will end up with compressed archives (tgz and zip) in the `migration/orion-to-tessera/build/distribution` directory so you would need to add instructions to expand them.
To do so, at the root of the project, run `./gradlew clean migration:orion-to-tessera:installDist -x test`
- `./gradlew` uses the wrapper
- `clean` purges all previous build artefacts
-  `migration:orion-to-tessera:` is the module that you want to target and only it
- `installDist` is the build task that will assemble a binary and expand the folders in `migration/orion-to-tessera/build/install/migration/` with `bin` and `lib` subfolders directly ready to run `./bin/migrate`.
- `-x test` will exclude the tests as we only need to build the binaries, not run the whole test suite that takes ages...

Comment (alexandratran) [reply docs/Tutorials/Migrating-to-Tessera.md]: I have Java JDK v15 and `./gradlew clean migration:orion-to-tessera:installDist -x test` did not work with the default gradle v6.0.1 but it did work after editing `tessera/gradle/wrapper/gradle-wrapper.properties` to use v6.3.

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
1. [Build](#build-migration-utility) or [download](#download-migration-utility) the migration utility.
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
3. Perform [configuration and database migration](#migrate).
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
4. Start Tessera with the new configuration and database files.
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
2. Shut down the Orion and Hyperledger Besu nodes.
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
5. Start Hyperledger Besu nodes.
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
The utility can be built from the [Tessera repository](https://github.com/ConsenSys/tessera/).
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
First, clone the Tessera repository:
```

For the rest of this section, I suggest removing the list numbers and wrapping the terminal commands in bash [code blocks](https://github.com/ConsenSys/doc.common/wiki/MkDocs-And-Custom-Markdown-Guide#including-code-in-the-documentation).

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
A utility is included in Tessera which enables migration of an Orion configuration
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
file and database to a Tessera configuration file and database. No changes are required to the Besu configuration file to migrate.
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]: `./tessera/migration/orion-to-tessera/build/install/migrate/bin/migrate` for me

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
By default Tessera uses an H2 database. However, you can configure alternative databases.
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
Refer to the [SQL Data Definition Language files](https://github.com/consensys/tessera/tree/master/ddls/create-table)
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
    If migrating from an SQL database to Tessera, you must add the JDBC driver
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
    to the `CLASSPATH` environment variable and
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
    the start script at `tessera/migration/orion-to-tessera/build/install/migrate`.
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
    Password protected keys are renamed to `.orion` on migration and added to the Tessera configuration file.
```

"config" should be changed to "configuration" when not in backticks throughout the doc.

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (vmichalik) [reply docs/Tutorials/Migrating-to-Tessera.md]: blocked by hyperledger/besu-docs/issues/499

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (vmichalik) [reply docs/Tutorials/Migrating-to-Tessera.md]: thank you for the example and detail

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
Build the migration utility with the Gradle wrapper `gradlew`:
```

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]: I have Java JDK v15 and this did not work with the default gradle v6.0.1, but did work when I changed it to v6.3 in the gradle wrapper properties file.

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]: I suggest removing the list numbers from this section and adding code block examples or link to examples on each item.

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]: Add blank line above this header to fix markdownlint error.

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]: Add blank line above this to fix markdownlint error.

File: docs/Tutorials/Migrating-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-to-Tessera.md]:
```suggestion
Navigate to the project root directory:
```

**Review by alexandratran — COMMENTED** [review]
Added comments.

File: docs/Tutorials/Migrating-from-Orion-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-from-Orion-to-Tessera.md]:
```suggestion
=== "Command"
```

**Review by NicolasMassart — COMMENTED** [review]
Note to other reviewers @alexandratran and @macfarla  :
This PR will be good to me once your remaining comments will be dealt with.
So when you are satisfied with it too, please approve and merge, it only requires one approval.

File: docs/Tutorials/Migrating-from-Orion-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-from-Orion-to-Tessera.md]: When will this download link be available? Not sure we should merge with a broken link.

File: docs/Tutorials/Migrating-from-Orion-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-from-Orion-to-Tessera.md]:
```suggestion
Begin the migration by running `migrate` with all required options.
```

File: docs/Tutorials/Migrating-from-Orion-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-from-Orion-to-Tessera.md]:
```suggestion
On a successful migration, the count of transactions and privacy groups migrated will match expected values.
```

File: docs/Tutorials/Migrating-from-Orion-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-from-Orion-to-Tessera.md]:
```suggestion
Start Besu and Tessera using the new Tessera configuration file and database.
```

File: docs/Tutorials/Migrating-from-Orion-to-Tessera.md
Comment (alexandratran) [inline docs/Tutorials/Migrating-from-Orion-to-Tessera.md]:
```suggestion
Verify the state on your Besu and Tessera nodes using `priv_debugGetStateRoot` and `priv_getTransactionCount`; it should be identical to the results from earlier.
```

File: docs/Tutorials/Migrating-from-Orion-to-Tessera.md
Comment (macfarla) [inline docs/Tutorials/Migrating-from-Orion-to-Tessera.md]: Password protected keys are migrated to the Tessera format as part of the migration. The original (Orion format) keys are renamed with `.orion` suffix.

File: docs/Tutorials/Migrating-from-Orion-to-Tessera.md
Comment (vmichalik) [reply docs/Tutorials/Migrating-from-Orion-to-Tessera.md]: nope, I am trying to get a clear date on the link availability

File: docs/Tutorials/Migrating-from-Orion-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-from-Orion-to-Tessera.md]:
```suggestion
=== "Request"
```

File: docs/Tutorials/Migrating-from-Orion-to-Tessera.md
Comment (NicolasMassart) [inline docs/Tutorials/Migrating-from-Orion-to-Tessera.md]:
```suggestion
=== "Request"
```

**Review by alexandratran — APPROVED** [review]
Looks good to me.

**Pull request merged**
